packages/calypso-e2e/src/lib/pages/me/account-settings-page.ts

```diff
@@ -43,13 +55,22 @@ export class AccountSettingsPage {
 		// The only thing that doesn't appear until all the loading is done is the sidebar of items to be deleted.
 		// So we must wait for that text before continuing, or our close account button click can get swallowed!
 		await this.page.waitForSelector( selectors.deletedItemsSidebar );
+
+		// Ensure the button is not `Manage Purchases` but rather `Close Account` a few times.
+		await reloadAndRetry( this.page, waitForPurchasesRemoved );
+
+		// `Close Account` button on /me/account/close.
 		await this.page.click( selectors.closeAccountButton );
```

Comment:
Optional but just wanted to double check: is there any reason we shouldn't just wait for the button we _are_ looking for? Or would that not work in this case?

I was just thinking of Bart's recommendation to not wait on anything that you don't need to directly interact with, and I wasn't sure if we could apply that in this case!

Reply:
Good question - no particular reason for waiting on an element that should not be there, instead of looking for an element that we want to be there.

I'll revise in an upcoming commit, need to resolve a conflict too.